semgrep-core/src/core/il/IL.ml

```diff
@@ -275,7 +275,7 @@ and call_special =
    | StringAccess of string (* for records/hashes *)
 *)
 and anonymous_entity =
-  | Lambda of function_definition
+  | Lambda of G.function_definition * function_definition
```

Comment:
Why do we need to keep the original G.function_definition here?

Reply:
It's needed so that later on, when we enter into the scope of the lambda, we can properly add the arguments into the environment. The original code operated only on `G.function_definition`, and I didn't understand it well enough to adapt it, so I just copied it here. Is it that bad to carry both?

Reply:
Maybe add a (* TODO? we probably should drop the G.function_definition and refactor the code elsewhere
to operate on a function_definition instead of G.function_definition *)